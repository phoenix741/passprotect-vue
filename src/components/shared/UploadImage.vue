<template lang="pug">
v-list-tile#add-image(v-on:click="addImage()")
  v-list-tile-title
    slot
    input(ref="fileInput",type="file",accept="image/*",@change="onChange",hidden)
</template>

<script>
export default {
  props: {
    value: String
  },
  data () {
    return {
      base64: this.value
    }
  },
  methods: {
    addImage () {
      this.$refs.fileInput.click()
    },
    onChange (event) {
      if (event.target.files && event.target.files[0]) {
        let file = event.target.files[0]
        let reader = new FileReader()
        reader.addEventListener('load', e => {
          const src = e.target.result
          let [, base64] = src.split(',')
          this.$emit('input', base64)
        })
        reader.readAsDataURL(file)
      }
    }
  }
}
</script>

