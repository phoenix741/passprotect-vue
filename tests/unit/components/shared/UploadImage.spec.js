import { shallowMount } from '@vue/test-utils'
import UploadImage from '@/components/shared/UploadImage.vue'

describe('UploadImage.vue', () => {
  it('Test that the component show a progress bar', () => {
    const UploadImageComponent = shallowMount(UploadImage, {
      propsData: {
        value: 'base64'
      }
    })
    expect(UploadImageComponent.element).toMatchSnapshot()
  })
})
