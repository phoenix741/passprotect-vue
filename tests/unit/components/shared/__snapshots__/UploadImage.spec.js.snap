// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadImage.vue Test that the component show a progress bar 1`] = `
<vlisttile-stub
  activeclass="primary--text"
  id="add-image"
>
  <vlisttiletitle-stub>
    <input
      accept="image/*"
      hidden="hidden"
      type="file"
    />
  </vlisttiletitle-stub>
</vlisttile-stub>
`;
